### DISCUSSION TITLE
Failing tests!

### DISCUSSION THREAD

**Discussion by smpeyres** [shannon-lab/zapdos#128] [Q&A Getting Started] [answered]
I updated MOOSE, and all of my tests for MOOSE pass, but now I'm getting some tests failing when I try to compile ZAPDOS. The errors are too long to paste here, so I attached them via a text file. 

Edit: I am using the wrong words: I meant compiling but said tests instead. 

Are there some new additions to the MOOSE framework that ZAPDOS hasn't dealt with yet? If so, how can I fetch an older version of MOOSE for the time being?

[errors.txt](https://github.com/shannon-lab/zapdos/files/8850105/errors.txt)

**Comment by csdechant** [comment, marked as answer]
Could you please elaborate on what you mean by "updated MOOSE". Based on `errors.txt`, the error occurs when building MOOSE as a submodule. What is the version of MOOSE you are using as a submodule? If you enter `git submodule status`, it should return:
`f84d6a63256846fa95d55e4859875bc4280a974a crane (v0.1.0-101-gf84d6a6)`
`ec3499546d443c3bdf384a7eb812f806d25f47bb moose (snapshot-20-10-27-26928-gec3499546d)`
`fc305d6fc10b5c96179c774eb73f9b71221fdf8a squirrel (heads/master)`
If you have difference version of the submodule, then your current version of Zapdos might be outdated and you would need to rebase your branch.

**Reply by smpeyres** [reply]
1. I ran
**mamba update --all
cd ~/projects/moose
git fetch origin
git rebase origin/master**
as my "update MOOSE"

2. I thought of the submodule after noticing the errors occurred in zapdos/moose/... so I performed
**git fetch origin
git rebase origin/master**
while in zapdos/moose and the compiling turned out just fine and all the ZAPDOS tests passed. But let me know if this is the absolute wrong thing to do!

This makes me thing the MOOSE submodule was out of date, in contrast to your suggestion.

**Reply by csdechant** [reply]
Rebasing in a submodule directory is usually the last step to take because it will only update that submodule. The first step is to check if the submodules are out-of-date compared to the Github repository and rebase the branch, since this will update everything (including the submodules). If there is still an error, then go ahead and rebase within the submodule in question. The difference is that rebasing outside the submodule updates to the current version on the Zapdos repository and rebasing inside the submodule updates to the current version on the MOOSE repository. In this case, it was that the MOOSE version on the Zapdos repository is outdated, due to libmesh being updated 17 hours ago. I will make a quick PR to fix the Zapdos repository.
